src/main/java/org/mskcc/cbio/oncokb/service/SlackService.java

```diff
@@ -399,85 +398,54 @@ public String getOptionValueArgument(String value) {
         return values[0];
     }
 
-    public boolean withTrialAccountNote(UserDTO userDTO, ActionId actionId) {
-        if (
-            ObjectUtil.isObjectEmpty(userDTO.getAdditionalInfo())
-                || userDTO.getAdditionalInfo().getTrialAccount() == null
-                || userDTO.getAdditionalInfo().getTrialAccount().getActivation() == null
-        ) {
+    public boolean withNote(MailType mailType, UserDTO userDTO, ActionId actionId) {
```

Comment:
created a general withNote function that checks if a certain MailType has been sent/was just sent. this works for the regular emails we wish to send (anything through the slack view modal). it will not work for automated ones as those have their own logic. so if we plan to add more automated emails the logic will still need to be changed here.